### DISCUSSION TITLE
Approach of invalidating render preview cache

### DISCUSSION THREAD

**Discussion by nickolay-kondratyev** [dendronhq/dendron#1342] [General]
Approach for render preview cache. 

[[Fixing Caching Preview with References|scratch.2021.09.15.103955.fixing-caching-preview-with-references]]

https://github.com/dendronhq/dendron-site/commit/f2c039d676c398df4d141ad0ef2777bff7db75cc?branch=f2c039d676c398df4d141ad0ef2777bff7db75cc&diff=unified&short_path=e2b2734#diff-e2b273433bb525b7c990459979df73c78e94c47dc6d5d77f4b2332a2a5140517

**Comment by kevinslin** [comment]
> Do we have constraints that a file name is unique within a workspace? To allow us to index by fname without a vault. 

No we do not. I don't see us enforcing that either. If a vault name isn't given with a link, then it is ambiguous. Only [Cross vault links](https://wiki.dendron.so/notes/3472226a-ff3c-432d-bf5d-10926f39f6c2.html#cross-vault-links) are unambgiguously defined for multi-vault. 

This is okay though -> our default resolution when no xvault link is specified is to use the first vault that has a matching file. Since this is how it is rendered, we can use [[Approach 2: traverse the links down from rendered note|scratch.2021.09.15.103955.fixing-caching-preview-with-references#approach-2-traverse-the-links-down-from-rendered-note]] to see if the ref has been updated. 

My vote is for the traversal approach (when previewing a note, check if note refs have been updated)

**Reply by nickolay-kondratyev** [reply]
Unfortunately I didn't sync the most up to date version of the doc (since it was in a different vault, seeds/dendron.dendron-site). Which had a line in approach #2: **Giving more thought this approach isn't going to work since we render the backlinks in the preview.**. If we just look down through the direct links the backlinks will still be stale.

**Comment by SeriousBug** [comment]
Looks like this caching is also causing issues with hover previews:

Note displayed in hover preview:
![](https://i.imgur.com/NuUxgll.png)

Actual note contents:
![](https://i.imgur.com/ep3vK0W.png)

Hover previews are generated using a reference: we run the processor with a "fake" note that has the contents `![[link-target]]`, which renders the note preview for us. If the caching is working with timestamps, this could be fixed by setting the timestamp of the fake note to the timestamp of the target note.

Since a workaround is possible, it's okay to not fix this particular issue when fixing the caching. But I'll hold off on trying to fix the preview to see how this issue pans out.
